src/common/translation.cpp

```diff
@@ -168,7 +160,7 @@ wxString GetPreferredUILanguage(const wxArrayString& available)
     if (!langNoMatchRegion.empty())
         return langNoMatchRegion;
 
-    return GetPreferredUILanguageFallback(available);
+    return wxString();
```

Comment:
I was always very confused by the "locale-based as a fallback" code, but as I didn't understand it, I didn't change it (or at least didn't do it intentionally) as I was too afraid to remove it and break something. If we can remove it, this is great, as it makes things simpler to understand, but I can't help wondering how could we have it for all these years without anybody knowing what it was for...

Reply:
Reconstructing from git blame, IIRC the original could get a list of languages only on some platforms (OS X?) and maybe it could fail, hence fallback. The "fallback" implementation was definitely used as the only one on UNIX originally.

The wrong use the fallback, i.e. as the last thing, was a bug I implemented at the beginning, but it didn't demonstrate itself until these API changes. Possibly it was that even older code just returned locale-based single language, so falling through to older logic made sense. In any case, now that we get a list of languages on every platform, there's no point.

Reply:
Thanks for digging into this!